streams/src/test/java/org/apache/kafka/streams/state/internals/TimeOrderedCachingPersistentWindowStoreTest.java

```diff
@@ -138,51 +144,69 @@ public void setUp() {
 
     @After
     public void closeStore() {
-        cachingStore.close();
+        try {
+            cachingStore.close();
+        } catch (final RuntimeException runtimeException) {
+           /*
+           It will reach here for the testcases like
+           shouldCloseCacheAndWrappedStoreAfterErrorDuringCacheFlush(),
+           shouldCloseWrappedStoreAfterErrorDuringCacheClose(),
+           shouldCloseCacheAfterErrorDuringStateStoreClose()
+
+           In these testcases we have set the doThrow for the cache#close
+           cache#flush, underlyingStore#close method.
+            */
+        }
+
     }
 
     @SuppressWarnings("deprecation")
     @Test
     public void shouldDelegateDeprecatedInit() {
-        final RocksDBTimeOrderedWindowStore inner = EasyMock.mock(RocksDBTimeOrderedWindowStore.class);
-        EasyMock.expect(inner.hasIndex()).andReturn(hasIndex);
-        EasyMock.replay(inner);
-        final TimeOrderedCachingWindowStore outer = new TimeOrderedCachingWindowStore(inner, WINDOW_SIZE, SEGMENT_INTERVAL);
+        final RocksDBTimeOrderedWindowStore inner = mock(RocksDBTimeOrderedWindowStore.class);
+        when(inner.hasIndex()).thenReturn(hasIndex);
+
+        final TimeOrderedCachingWindowStore outer =
```

Comment:
I think we do not need to spy on the outer, since here we are only checking that when `outer.init` is called, its `inner.init` is called too. Ditto for other tests